includes/functions.php

```diff
@@ -1158,9 +1171,7 @@ function yourls_get_all_options() {
  */
 function yourls_update_option( $option_name, $newvalue ) {
 	global $ydb;
-
-    $option = new \YOURLS\Database\Options($ydb);
-    $update = $option->update($option_name, $newvalue);
+    $update = $ydb->yourls_options->update($option_name, $newvalue);
```

Comment:
@LeoColomb opinion on this? ie using the same `yourls_options` instead of making a new `Options($ydb)` each time? Useful? Useless? Meaningless optimization?

Reply:
Just weird! 😆
No, more seriously, it won't change anything.

Let's recap.

```php
new \YOURLS\Database\Options($ydb);
```
basically return `$ydb` wrapped up with some functions/methods:

https://github.com/YOURLS/YOURLS/blob/575ded5925352547ea498b93d888f19d3ba1197d/includes/Database/Options.php#L32-L34

In other words,

```php
$ydb->yourls_options = new \YOURLS\Database\Options($ydb);
```
is more or less equivalent to
```php
$ydb->yourls_options = $ydb; # <--- ouch
$ydb->yourls_options->update = function (...) {...}
```
which is not really useful.
And you still need to use the `global` (which is not really a best practice! 😅).

In terms of functionality, it's equivalent.
In terms of readability, opinion can differ I think.
In terms of brain visualization, it doesn't help at all.

---

Without your addition, a new object is created and destroyed at each test, but `$ydb` keeps the updates.
Then in terms of memory, it's almost equivalent.
And in terms of CPU, yaaaah, we might win some micro seconds...

---

At the end, to be fair, if your prefer this new version go ahead. But yeah, it's meaningless optimization.

Reply:
OK, I'll remove this then, thanks for your feedback. What you describe is what I thought, just wanted your input to be sure.